testing/dnm: limit compaction to snapshot offset

bharathv opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

bharathv commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--ea0f6f83-66c8-42fd-9ca1-848fc3aa3db7-->
<!--TEST RESULTS-->
<!--fb062112-26bc-4f10-ad9b-379f01d5f18c-->
<!--TEST RESULTS-->
<!--382544d1-fae1-4253-abe3-b4ac18f72375-->
<!--TEST RESULTS-->
<!--323e6c54-1c0c-4ab3-b3bd-d2b95afc8005-->
<!--TEST RESULTS-->
<!--d9c39d2a-69a8-4563-a897-bb8c693dc8ec-->
<!--TEST RESULTS-->
<!--334e79ad-4dad-4a2d-914e-f8d3aae228b7-->
<!--TEST RESULTS-->
<!--62660ec8-62b6-47d7-a61c-3838db760928-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75441">build#75441</a></summary>


| test_class                                           | test_method                                | test_arguments                                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history                                                                                                                                                                               |
|:-----------------------------------------------------|:-------------------------------------------|:----------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest                    | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bf9-bd90-4243-81b0-3f8bbf0fb1c9 | FAIL          | 0/70     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest                    | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bf9-bd90-4243-81b0-3f8bbf0fb1c9 | FAIL          | 0/70     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest                    | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bf9-bd90-4243-81b0-3f8bbf0fb1c9 | FAIL          | 0/70     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest                    | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bf9-bd90-4243-81b0-3f8bbf0fb1c9 | FAIL          | 0/70     |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| test_with_basic_consume_fixture/BasicConsumerFixture | TestBasicLeadershipTransfer/2              |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bca-a653-4b0a-9d29-2aca0b3f3adb | FLAKY         | 1/2      |          |                                                                                                                                                                                            |
| test_with_basic_consume_fixture/BasicConsumerFixture | TestBogusPartitionIds/0                    |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bca-a653-4b0a-9d29-2aca0b3f3adb | FLAKY         | 1/2      |          |                                                                                                                                                                                            |
| test_with_basic_consume_fixture/BasicConsumerFixture | TestBogusPartitionIds/2                    |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bca-a653-4b0a-9d29-2aca0b3f3adb | FLAKY         | 1/2      |          |                                                                                                                                                                                            |
| test_with_basic_consume_fixture/BasicConsumerFixture | TestUnassignTopic/1                        |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bca-a653-4b0a-9d29-2aca0b3f3adb | FLAKY         | 1/2      |          |                                                                                                                                                                                            |
| test_with_basic_consume_fixture/BasicConsumerFixture | TestUnassignTopic/2                        |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75441#019a3bca-a653-4b0a-9d29-2aca0b3f3adb | FLAKY         | 1/2      |          |                                                                                                                                                                                            |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75454">build#75454</a></summary>


| test_class                        | test_method                                | test_arguments                                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                      | test_history                                                                                                                                                                               |
|:----------------------------------|:-------------------------------------------|:----------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41d7-7a9c-4a80-a8fc-0789293308d3 | FLAKY         | 69/70    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41f4-b6e1-45e3-a3b2-2a7a09f39489 | FLAKY         | 67/70    | upstream reliability is '100.0'. current run reliability is '86.95652173913044'. drift is 13.04348 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41f4-b6e1-45e3-a3b2-2a7a09f39489 | FLAKY         | 67/70    | upstream reliability is '100.0'. current run reliability is '86.36363636363636'. drift is 13.63636 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41d7-7a9c-4a80-a8fc-0789293308d3 | FLAKY         | 66/70    | upstream reliability is '100.0'. current run reliability is '82.6086956521739'. drift is 17.3913 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41f4-b6e1-45e3-a3b2-2a7a09f39489 | FLAKY         | 66/70    | upstream reliability is '100.0'. current run reliability is '81.81818181818183'. drift is 18.18182 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75454#019a41f4-b6e1-45e3-a3b2-2a7a09f39489 | FLAKY         | 69/70    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75478">build#75478</a></summary>


| test_class                        | test_method                                | test_arguments                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                      | test_history                                                                                                                                                                               |
|:----------------------------------|:-------------------------------------------|:-----------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}               | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4212-432d-915f-ad790a3025f5 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}               | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4213-45b9-93b0-edf21ce612b1 | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}               | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-859e-4e7b-9545-3a7f84cf8dbf | FLAKY         | 27/30    | upstream reliability is '100.0'. current run reliability is '86.36363636363636'. drift is 13.63636 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4212-432d-915f-ad790a3025f5 | FLAKY         | 27/30    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4212-444d-86f3-bfc5f04eafed | FLAKY         | 27/30    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4213-48a9-9b29-1418e24cef43 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-4213-4bca-9b06-449916dae5b1 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-859e-4e7b-9545-3a7f84cf8dbf | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"} | integration | https://buildkite.com/redpanda/redpanda/builds/75478#019a4a27-859f-4730-9b62-12a3b9a99c9c | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75494">build#75494</a></summary>


| test_class                        | test_method                                | test_arguments                                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     | test_history                                                                                                                                                                               |
|:----------------------------------|:-------------------------------------------|:----------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b29-ef8b-43b2-b227-ba896f8b8388 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b29-ef8c-43a4-88ea-6d268573a7f5 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b29-ef8c-45e2-9e20-d0a7def24b91 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All aborts"}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b32-579c-42c3-a7b3-b732c54db1a7 | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b32-579a-4d4f-8fa4-a3f21b840c39 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b32-579c-42c3-a7b3-b732c54db1a7 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b32-579c-4684-9ae2-676a42f1b13e | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b29-ef8d-49d5-960e-5884714cfef9 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b32-579b-4ef2-8af5-bc96d142e9cc | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.9090909090909'. drift is 9.09091 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b44-f25e-4fb9-8cfe-6bd33653a914 | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75494#019a4b29-ef8d-49d5-960e-5884714cfef9 | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75502">build#75502</a></summary>


| test_class                              | test_method                                | test_arguments                                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                               |
|:----------------------------------------|:-------------------------------------------|:----------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest       | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75502#019a4bb9-7bca-41d1-9856-43010d3ec6d1 | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest       | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "All commits"}                         | integration | https://buildkite.com/redpanda/redpanda/builds/75502#019a4bc0-9ca6-468a-9914-2666c0b906df | FLAKY         | 28/30    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest       | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Mixed aborts and commits"}            | integration | https://buildkite.com/redpanda/redpanda/builds/75502#019a4bb9-7bca-41d1-9856-43010d3ec6d1 | FLAKY         | 26/30    | upstream reliability is '100.0'. current run reliability is '81.81818181818183'. drift is 18.18182 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest       | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75502#019a4bc0-9ca3-45c8-881f-0163d738f2c7 | FLAKY         | 27/30    | upstream reliability is '99.2248062015504'. current run reliability is '86.36363636363636'. drift is 12.86117 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| LogCompactionTxRemovalUpgradeTest       | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75502#019a4bc0-9ca6-468a-9914-2666c0b906df | FLAKY         | 29/30    | upstream reliability is '99.22779922779922'. current run reliability is '95.23809523809523'. drift is 3.9897 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
| src/v/cluster/tests/tx_compaction_tests | src/v/cluster/tests/tx_compaction_tests    |                                                           | unit        | https://buildkite.com/redpanda/redpanda/builds/75502#019a4b82-b436-4fe9-8909-b8ac9d96564c | FAIL          | 0/1      |                                                                                                                                                                        |                                                                                                                                                                                            |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75515">build#75515</a></summary>


| test_class                        | test_method                                | test_arguments                                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                               |
|:----------------------------------|:-------------------------------------------|:----------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| LogCompactionTxRemovalUpgradeTest | test_tx_control_batch_removal_with_upgrade | {"test_case_name": "Multi-segment spanning transactions"} | integration | https://buildkite.com/redpanda/redpanda/builds/75515#019a4c94-a3e6-4184-bb01-6a28dbf7aa62 | FLAKY         | 28/30    | upstream reliability is '99.29078014184397'. current run reliability is '90.47619047619048'. drift is 8.81459 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalUpgradeTest&test_method=test_tx_control_batch_removal_with_upgrade |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75518">build#75518</a></summary>


| test_class                           | test_method                     | test_arguments                                                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                  |
|:-------------------------------------|:--------------------------------|:------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkConsumeGroupsMirroringTest | test_continuous_group_sync      | {"source_cluster_spec": {"cluster_type": "redpanda"}, "with_failures": true}                                      | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ebc-432c-9869-f6257ab0f9bf | FLAKY         | 20/21    | upstream reliability is '98.78048780487805'. current run reliability is '95.23809523809523'. drift is 3.54239 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkConsumeGroupsMirroringTest&test_method=test_continuous_group_sync |
| DatalakeE2ETests                     | test_topic_lifecycle            | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1}                                                            | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ebe-45cc-821d-f366dc6b5fb4 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeE2ETests&test_method=test_topic_lifecycle                           |
| DatalakeTableNameTest                | test_topic_name_dot_replacement | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1}                                                          | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecd-408d-b6f6-f0fea69292de | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeTableNameTest&test_method=test_topic_name_dot_replacement           |
| MountUnmountIcebergTest              | test_simple_remount             | {"cloud_storage_type": 1}                                                                                         | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec5-457c-9338-c659a9731c38 | FLAKY         | 13/21    | upstream reliability is '100.0'. current run reliability is '61.904761904761905'. drift is 38.09524 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                     |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 10}      | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecb-4b77-a4b1-531c5e9d0ae7 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 131072}  | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecd-408d-b6f6-f0fea69292de | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 3}       | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec3-4936-aaa2-548aafc7d19a | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 10}     | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec4-48d5-a30c-994ddda5fcbc | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 131072} | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec5-457c-9338-c659a9731c38 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact,delete", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 3}      | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec6-474c-941a-c4ccccc42b19 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 10}             | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec7-4313-8c2e-586445cddd01 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 131072}         | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec8-4199-a622-4e3891386d08 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 100, "storage_compaction_key_map_memory_kb": 3}              | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec9-4466-aec1-22b0536264dd | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 10}            | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3eca-4144-82ab-09bd942bcde3 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 131072}        | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecb-4b77-a4b1-531c5e9d0ae7 | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| LogCompactionTest                    | compaction_stress_test          | {"cleanup_policy": "compact", "key_set_cardinality": 1000, "storage_compaction_key_map_memory_kb": 3}             | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecd-408d-b6f6-f0fea69292de | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTest&test_method=compaction_stress_test                        |
| PartitionMovementTest                | test_empty                      | {"num_to_upgrade": 0}                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecb-4b77-a4b1-531c5e9d0ae7 | FLAKY         | 17/21    | upstream reliability is '85.19283746556474'. current run reliability is '80.95238095238095'. drift is 4.24046 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                                |
| PartitionMovementTest                | test_static                     | {"num_to_upgrade": 0}                                                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3eb6-4c04-a613-6d19eccc227a | FLAKY         | 14/21    | upstream reliability is '86.34658664666166'. current run reliability is '66.66666666666666'. drift is 19.67992 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_static                               |
| ScalingUpTest                        | test_fast_node_addition         | null                                                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ec3-4936-aaa2-548aafc7d19a | FLAKY         | 18/21    | upstream reliability is '98.91304347826086'. current run reliability is '85.71428571428571'. drift is 13.19876 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ScalingUpTest&test_method=test_fast_node_addition                           |
| TxUpgradeCompactionTest              | upgrade_with_compaction_test    | null                                                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/75518#019a4cd8-3ecd-408d-b6f6-f0fea69292de | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TxUpgradeCompactionTest&test_method=upgrade_with_compaction_test            |
</details>
<!--/body-->

bharathv commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv committed:
stm: introduce tx_snapshot_offset() and last_snapshot_offset() utils

bharathv committed:
cluster/debug: plumb max_tombstone_removable_offset and snapshot offset

.. into the partition debug dump.

bharathv committed:
storage/compaction: clamp tombstone remove offset to snapshot offset

This ensures that the when the stm is reconstructed from snapshot that
cuts through a transaction, the recovery sees the entire transaction and
the control batch is not removed by transaction leaving it dangling.

bharathv committed:
compaction/tests: deflake log_compaction_test

bharathv commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

WillemKauf commented:
Force pushed to:

* resolve merge conflict post https://github.com/redpanda-data/redpanda/pull/28283

WillemKauf commented on tests/rptest/tests/log_compaction_test.py:
```diff
@@ -678,6 +679,9 @@ def __init__(self, test_context):
             "log_compaction_interval_ms": 1000,
             "log_segment_size": 2 * 1024**2,  # 2 MiB
             "compacted_log_segment_size": 1024**2,  # 1 MiB
+            # Trigger tombstone removal quickly
```
comment here could be a _tiny_ bit more colorful- without context and reading this commit in isolation, a reader won't understand what `storage_target_replay_bytes` & `log_segment_ms` may have to do with compaction of tombstones/tx batches

WillemKauf commented on tests/rptest/tests/log_compaction_test.py:
```diff
@@ -726,12 +732,104 @@ def check_tx_batches(self):
+            )
+        return all(node_results)
+
+    def wait_until_stms_caught_up(self):
```
This seems useful maybe in `utils` or elsewhere?

WillemKauf commented:
/ci-repeat 5
dt-repeat=10
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv committed:
debug expiration

bharathv commented:
/ci-repeat 5
dt-repeat=10
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv commented:
/ci-repeat 5
dt-repeat=10
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv commented:
/ci-repeat 5
dt-repeat=10
tests/rptest/tests/log_compaction_test.py::LogCompactionTxRemovalUpgradeTest.test_tx_control_batch_removal_with_upgrade

bharathv commented:
/ci-repeat 1
skip-redpanda-build

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75518-->
<!--df2aa24d-e3ac-4c76-8232-4274ae609728-->
<!--25dbc5fd-0065-4040-82da-2b993bf3c672-->
<!--2c92743d-e97f-4a9f-9c66-e6aea93ce0a5-->
<!--1c43b315-8791-48eb-aff2-24bce57327cd-->
<!--0638f1b3-cf13-4c01-a1c5-fa777ae3545f-->
<!--f9959f5a-a5e6-4763-b7bc-0b9f98f3880e-->
<!--48f61515-1f6b-4464-9a91-30570d38acf4-->
<!--5c3db0cd-6aab-415e-ac5a-90d4397c9241-->
<!--6845aba3-b568-4357-a16b-407189d8241e-->
<!--83796bd6-cdd0-4f76-b05b-84fabc272fef-->
<!--title-->
## Retry command for [Build#75518](https://buildkite.com/redpanda/redpanda/builds/75518)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":3}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":10}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":131072}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":10}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":3}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":10}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":10}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":131072}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":3}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":100,"storage_compaction_key_map_memory_kb":131072}
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":3}
tests/rptest/transactions/tx_upgrade_test.py::TxUpgradeCompactionTest.upgrade_with_compaction_test
tests/rptest/tests/log_compaction_test.py::LogCompactionTest.compaction_stress_test@{"cleanup_policy":"compact,delete","key_set_cardinality":1000,"storage_compaction_key_map_memory_kb":131072}
```
<!--/body-->
